src/stores/provider.js

```diff
@@ -0,0 +1,112 @@
+    const nameB = sourceObjectB.source_name.toUpperCase()
+    return nameA.localeCompare(nameB)
+  })
+}
+
+export const createUseProviderStore = (services = providerServices) =>
+  defineStore('provider', () => {
+    /**
+     * @type {{fetchState: Record<import('../store/types').SupportedMediaType, import('../store/types').FetchState>, providers: Record<import('../store/types').SupportedMediaType, import('../store/types').MediaProvider[]>}}
+     */
+    const state = reactive({
+      providers: {
+        [AUDIO]: [],
+        [IMAGE]: [],
+      },
+      fetchState: {
+        [AUDIO]: {
+          isFetching: false,
+          fetchingError: null,
+        },
+        [IMAGE]: {
+          isFetching: false,
+          fetchingError: null,
+        },
+      },
+    })
+
+    /**
+     * Returns the display name for provider if available, or capitalizes the given providerCode.
+     *
+     * @param {string} providerCode
+     * @returns {string}
+     */
+    const getProviderName = (providerCode) => {
+      const searchStore = useSearchStore()
+      const mediaType =
+        searchStore.searchType === ALL_MEDIA ? IMAGE : searchStore.searchType
+      const matchingProviders = state.providers[mediaType].filter(
+        (p) => p.source_name === providerCode
+      )
+      return matchingProviders.length
+        ? matchingProviders[0].display_name || capital(providerCode)
+        : capital(providerCode)
+    }
+
+    const fetchMediaProviders = async () => {
+      await Promise.all(
+        supportedMediaTypes.map((mediaType) =>
+          fetchMediaTypeProviders(mediaType)
+        )
+      )
+    }
+    /**
+     * Fetches providers for a set media type, and initializes the provider filters by calling the search store `initProviderFilters` method.
+     * @param {import('../store/types').SupportedMediaType} mediaType
+     * @returns {Promise<void>}
+     */
+    const fetchMediaTypeProviders = async (mediaType) => {
+      const searchStore = useSearchStore()
+      state.fetchState[mediaType] = {
+        fetchingError: null,
+        isFetching: true,
+      }
+      let sortedProviders =
+        /** @type {import('../store/types').MediaProvider[]} */ ([])
+      try {
+        const res = await services[mediaType].getProviderStats()
+        sortedProviders = sortProviders(res.data)
+      } catch (error) {
+        warn(`Error getting ${mediaType} providers: `, error)
+        state.fetchState[mediaType].fetchingError =
+          'response' in error ? error.response.code : error.message
+      } finally {
+        state.fetchState[mediaType].isFetching = false
+
+        state.providers[mediaType] = sortedProviders
+
+        searchStore.initProviderFilters({
+          mediaType,
+          providers: sortedProviders,
+        })
+      }
+    }
+    return {
+      state,
+      nuxtServerInit: fetchMediaProviders,
+      getProviderName,
+    }
+  })
+export const useProviderStore = createUseProviderStore()
```

Comment:
Why this `create...` pattern?

Reply:
When testing, we mock the `services`, and create the `provider` store using `const providerStore = createUseProviderStore(mockServices)`.
For the components, we can initialize by simply calling `useProviderStore`, which re-uses the store that has already been created using the default services.

Reply:
Could we mock the services in the tests using `jest.mock`?